go/database/mpt/verification_proof.go

```diff
@@ -0,0 +1,277 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package mpt
+
+import (
+	"context"
+	"fmt"
+	"github.com/Fantom-foundation/Carmen/go/common"
+	"github.com/Fantom-foundation/Carmen/go/common/interrupt"
+	"github.com/Fantom-foundation/Carmen/go/common/witness"
+	"golang.org/x/exp/maps"
+)
+
+// VerifyProofArchiveTrie verifies the consistency of witness proofs for an archive trie.
+// It reads the trie for each block of the archive and loads all accounts and storage slots.
+// Then, it extracts witness proofs for the same accounts and storage slots of each block.
+// Finally, the actual values from the trie are compared with the values extracted from the witness proofs.
+func VerifyProofArchiveTrie(ctx context.Context, dir string, config MptConfig, from, to int, observer VerificationObserver) (err error) {
+	trie, err := OpenArchiveTrie(dir, config, NodeCacheConfig{})
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = trie.Close()
```

Comment:
err is not checked here

Reply:
It is assigned to the output of this method. 

I do not very like this syntax, because I myself consider it confusing, but do not know if this can be done better in golang.

It is tested here: 
https://github.com/Fantom-foundation/Carmen/pull/947/files#diff-8bb57076b2dc35d3c0b28ecd9488c8c002f0408e571713e8462c7b7e8a6d06aeR106

Reply:
Oh you are right. I always miss this.

You can do something like this to make sure the reader sees the check, but its very unnecessary.
```suggestion
		closeErr := trie.Close()
		if closeErr != nil {
		   err = errors.Join(err, closeErr)
		}
```

Reply:
I think this code is rather weird - `err` is always `nil` and `errors.Join` checks for `nil` as well  - i.e. for me it would be puzzling what this whole construct is good for